### ISSUE TITLE
SAE Preview Window Crash

### ISSUE THREAD

**Issue by Kaio-Dev** [iperov/DeepFaceLab#334]
Left the model training with the "train SAE.bat"  and then went out for a few hour. 
But when i came back the preview had stopped responding.  Furthermore it forced closed.
Tried to re train but was greeted with 
![image](https://user-images.githubusercontent.com/28940326/62688016-48c83e00-ba0b-11e9-9194-7c7f030379c2.png)
Decided to convert the model and then convert it to mp4 but the footage was the same as the original. 

Further information

File in use:
7) convert SAE.bat :

```
@echo off
call _internal\setenv.bat

"%PYTHON_EXECUTABLE%" "%DFL_ROOT%\main.py" convert ^
    --input-dir "%WORKSPACE%\data_dst" ^
    --output-dir "%WORKSPACE%\data_dst\merged" ^
    --aligned-dir "%WORKSPACE%\data_dst\aligned" ^
    --model-dir "%WORKSPACE%\model" ^
    --model SAE
    
pause
```
OS:  Windows 10 Pro
Python version: 3.6.4

**Comment by Akanee31** [comment]
Try to move and run DFL folder on your C:/ (main hard drive)

**Comment by Kaio-Dev** [comment]
Hey, I moved it over and ran it from the new location. The problem still persists. 
![image](https://user-images.githubusercontent.com/28940326/62698665-027dd980-ba21-11e9-9b36-93806af7b701.png)

**Comment by Akanee31** [comment]
Rename your deepfacelab folder in DFL.
put the folder on c: (root).

**Comment by Kaio-Dev** [comment]
![image](https://user-images.githubusercontent.com/28940326/62766273-6a8df780-bad5-11e9-891e-024bbab20e26.png)
Not too many surprises there

**Comment by Auroir** [comment]
(First, move those files back if you haven't. Your folder should have `_internal`, `workspace`, and the `.bat` files in it. If you don't know how to revert what you changed then just unpack a new set of the tools and copy over your workspace.)

I think your model collapsed while you were gone, do you have the last preview image? Is it good? Try to enable the gradient clipping option, maybe it will recover.

If that doesn't fix it: 

1. Back up your saved model
2. Make a new model with the **exact** same parameters
3. Train it to at least 2 iterations then save
4. Copy all the .h5 files from your backup into the new model's folder, overwriting the new model's data.
5. Relaunch training. 

The idea here is to reset your .dat file which holds things like training history and meta-parameters but not the actual trained model.

**Comment by Kaio-Dev** [comment]
I don't have a preview window still available, but from when the model collapsed the preview looked okay. Gradient clipping didn't recover it.
Are the parameters of the config for the model training saved anywhere? As I don't remember them.
Thanks,

**Comment by Auroir** [comment]
Yeah, it looks like your model collapsed. RIP.

You can try to copy over individual .h5 files with the same steps and try to save as much data as possible, but I don't think it's worth your time. For the next run:
- Turn on clipping from the start, it's worth the tiny slowdown.
- Increase your batch size, the default when you select 0 is hard coded to 4 right now. Increase it until you get OOM errors at launch, then reduce it by 2 (you can try reducing by 1 but it might OOM later).
- Unless you only have 4GB of system RAM, use optimizer mode 2. It lets you use some system ram as vram so you can increase your batch size more. If you have 16 or 32GB you can try 3 but it's slow. YMMV. 
- Use CA weights. It makes your setup slow but you gain the time back during training, plus other benefits. It's good stuff.
- Triple check your faces. We still don't know why it failed. invalid faces is a common culprit.

**Comment by Auroir** [comment]
By the way, if you're making a new model now's the time to tune the dimensions to fit your GPU.

1. Make a new model, batch size 1
2. Train it to iteration 2, save, reload, and increase the batch size.
3. Repeat until you get OOM errors at launch. **As an example, say you get to 8** and it OOMs.
4. Delete your model and make a new one
5. During setup use 8 as your batch size (or whatever you got) and reduce the first 2 dimension prompts, reducing both by the same proportion (-5%, -15%, etc.) **Leave the 3rd prompt as default, it is auto-scaled based on the 2nd one but doesn't tell you. 3rd should be half of the 2nd, if not larger!**
6. Repeat, reducing the dimensions until you no longer get OOM errors with batch size 8 (or whatever you got)

(This is all LIAE, I don't remember if DF has 3 or 4 dimensions but the same idea applies)

**Comment by Kaio-Dev** [comment]
Hey, the issue still persists 
I clipped it happening this time.
https://youtu.be/Ctu7MV9OUUg
I'm not sure why the model is crashing as it is.

**Comment by ntmatter1** [comment]
For some reason since the last major Windows 10 update this seems to happen to me more and more - GTX 1080 Ti card and plenty of CPU and RAM.  Used to be able to run a batch size of 12 with higher dimensions than standard, but now I cannot run even the defaults.  Mainly getting "CUDA Unspecified Launch Failure" errors, sometimes others.

I've triad Auroir's steps above and it looks like it helps, particularly when I switched to liae.  Typically, the error is this:
2019-10-13 10:49:54.263699: E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure
2019-10-13 10:49:54.267340: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:274] Unexpected Event status: 1

But, it's still not perfect, I'll still get CUDA crashes even with a batch of 4, and especially when I am using the PC.  Used to never happen.  Are there other issues maybe related to the drivers or configuration of the graphics card itself that might be contributing?
